packages/core/src/screens/CredentialOfferAccept.tsx

```diff
@@ -84,8 +86,78 @@ const CredentialOfferAccept: React.FC<CredentialOfferAcceptProps> = ({ visible,
     if (credential.state === CredentialState.CredentialReceived || credential.state === CredentialState.Done) {
       timer && clearTimeout(timer)
       setCredentialDeliveryStatus(DeliveryStatus.Completed)
+
+      // Check if accepted credential has cached schema name
+      const existingMetadata = credential.metadata.get(AnonCredsCredentialMetadataKey)
+
+      if (existingMetadata?.schemaName) {
+        logger.debug('Accepted credential already has cached schema name', {
+          credentialId: credential.id,
+          schemaName: existingMetadata.schemaName,
+        })
+      } else {
+        logger.warn('Accepted credential missing cached schema name', {
+          credentialId: credential.id,
+          hasMetadata: !!existingMetadata,
+          schemaId: existingMetadata?.schemaId,
+        })
+
+        // Try to restore schema name from format data if available
+        const restoreSchemaMetadata = async () => {
+          try {
+            if (!agent?.credentials) {
+              return
+            }
+            const { offer } = await agent.credentials.getFormatData(credential.id)
+            const offerData = offer?.anoncreds ?? offer?.indy
+
+            if (
+              offerData &&
+              typeof offerData === 'object' &&
+              'schema_id' in offerData &&
+              offerData.schema_id &&
+              agent?.modules?.anoncreds
+            ) {
+              try {
+                const { schema: resolvedSchema } = await agent.modules.anoncreds.getSchema(offerData.schema_id)
+
+                if (resolvedSchema?.name) {
+                  logger.info('Restoring schema metadata for accepted credential', {
+                    credentialId: credential.id,
+                    schemaName: resolvedSchema.name,
+                    schemaId: offerData.schema_id,
+                  })
+
+                  // Store schema metadata
+                  const metadataToStore = {
+                    schemaId: offerData.schema_id,
+                    credentialDefinitionId: 'cred_def_id' in offerData ? offerData.cred_def_id : undefined,
```

Comment:
Is this hard coded property `cred_def_id` specific to `webhv` or a test credential you're working with. Another way, will every webhv credential have this propert?

Reply:
This comes from credo and is already a default value, not webvh sepcific:
https://github.com/openwallet-foundation/bifold-wallet/blob/165cb5388d2ec57ecdfd4baf0812cc5317101c3f/packages/core/src/screens/CredentialOffer.tsx#L130